src/librustc_mir/interpret/eval_context.rs

```diff
@@ -1466,7 +1466,7 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
 
     /// ensures this Value is not a ByRef
     pub(super) fn follow_by_ref_value(
-        &mut self,
+        &self,
```

Comment:
❤️

Reply:
Is there any way to get lints for "You used `&mut` but could have used `&`"?^^

Reply:
No. But I've been trying to write a lint for that since forever: https://github.com/rust-lang-nursery/rust-clippy/issues/353

Reply:
It should be easier on MIR because you can look for uses that aren't immutable reborrows.